collect_app/src/test/java/org/odk/collect/android/widgets/UrlWidgetTest.java

```diff
@@ -58,9 +59,9 @@ public void getAnswer_whenPromptHasAnswer_returnsAnswer() {
     }
 
     @Test
-    public void usingReadOnlyOption_makeAllClickableElementsDisabled() {
+    public void usingReadOnlyOption_doesNotShowUrlButton() {
         UrlWidget widget = createWidget(promptWithReadOnly());
-        assertThat(widget.getBinding().urlButton.getVisibility(), equalTo(View.GONE));
+        assertThat(widget.binding.urlButton.getVisibility(), equalTo(View.GONE));
```

Comment:
As discussed the URL widget should behave exactly the same in read only mode (as it is always "read only"). That being the case you shouldn't need any tests for read only for the widget.